Localmap

rgbkrk opened the pull request:
Not the best style or organization for being in an Electron app, but I wanted to get a prototype out.

![demo of localmap](https://cloud.githubusercontent.com/assets/836375/11450094/3e703300-9556-11e5-8800-bd5fdce5d190.gif)

Performance isn't so great here, will end up iterating on that. First with direct binary Blobs, second with WebGL.

rgbkrk committed:
canvas based localmap

should likely switch over to blobs and setting image.src or
using webgl

rgbkrk committed:
Clean up dependencies and onNext

kitten commented:
Great stuff :)

kitten commented:
I think the image drawing can be imrpoved a lot with this: https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas

Should I look into building a foundation in react for us?

kitten committed:
Improve rendering and several small fixes

- Convert received image to ImageData object
- Scale Canvas to window size
- Remove window border
- Rename bootstrap file to main
- Move application files to src folder

kitten commented:
@rgbkrk Pushed some small fixes and improvements

kitten committed:
Scale initially as well

kitten committed:
Add an arrow to indicate player orientation

kitten committed:
Add color modifier and set color to PipBoy-like color

kitten commented:
The new result is pretty decent I'd say.

![output](https://cloud.githubusercontent.com/assets/2041385/11451663/11faf260-95ce-11e5-90af-5f0eb27602e2.gif)

kitten commented:
@rgbkrk I just went ahead and merged this, cause it's not like we've got a lot of stuff here to worry about ^^

rgbkrk commented:
> @rgbkrk I just went ahead and merged this, cause it's not like we've got a lot of stuff here to worry about ^^

Totally cool! I would have loved to tear off my WIP tag but that doesn't matter. :smile:

rgbkrk commented on package.json:
```diff
@@ -29,12 +29,14 @@
   "homepage": "https://github.com/rgbkrk/pipboy#readme",
   "devDependencies": {
+    "babel-preset-es2015": "^6.1.18",
+    "electron-compile": "^1.0.1",
```
This will probably need to be a direct dependency as it's used in main, for when we package for wider distribution.

rgbkrk commented on index.html:
```diff
@@ -3,8 +3,42 @@
   </head>
   <body>
-    <h1>PipBoy running on Electron <script>document.write(process.versions.electron)</script></h1>
+    <canvas id="map"></canvas>
```
A+

rgbkrk commented on src/index.js:
```diff
@@ -0,0 +1,139 @@
+
+        const resize = Observable
+          .fromEvent(window, 'resize')
+          .merge(Observable.just())
```
Ah good, I figured we'd want to watch that. Thank you for making this awesome.

rgbkrk replied:
Wow and full screen is awesome. We'll want to make sure we keep the map at the right scale later.

rgbkrk commented:
Ok yeah you leveled this up :100:

rgbkrk commented:
> Should I look into building a foundation in react for us?

That would be wonderful.

rgbkrk commented:
> I think the image drawing can be imrpoved a lot with this: https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Pixel_manipulation_with_canvas

ah, `putImageData` is nice. I should probably use that in [bitjet](https://github.com/rgbkrk/bitjet) too. I allow for adjusting the width and height of each bit or byte represented in bitjet, but I'm guessing people don't need it -- they just need to be able to scale while investigating binary data.
